### PR TITLE
rebase-3-1-0-from-release-3-0-0

### PR DESCRIPTION
**Pull request by jack1902**
to: @ryandeivert @chunyong-lin 
cc: @airbnb/streamalert-maintainers
related to:
resolves:

## Background

Keeping `release-3-1-0` current with work carried out against `release-3-0-0` so that i can update the documentation accordingly

## Changes

* Carried out `git checkout release-3-1-0 && git pull` Ensure i have the latest branchs on my machine then `git checkout -b rebase_from/release-3-0-0 && git fetch upstream && git rebase upstream/release-3-0-0` This ensured i rebased a branch against the latest on the remote

* Updated the style of `rules.rst` when the `dynamic_outputs` feature was added

## Testing

* Ran `./tests/scripts/test_the_docs.sh` to ensure the documentation still built
* Ran `./tests/scripts/pylint.sh` Ensured pylint will pass
* Ran `./tests/scripts/unit_tests.sh` to ensure unit tests passed

### Notes to merger

* Add my remote `git remote add jack git@github.com:jack1902/streamalert.git && git fetch jack`

* Run `git merge -X theirs jack/rebase_from/release-3-0-0` to use my branch as the new `release-3-1-0` otherwise you'll run into all the merge conflicts i resolved when i did the rebase

### CODE REVIEW

**Comment by Ryxias** [comment]
We will need this ASAP since we have many changes in flight. Looks good so far, but there's 2 conflicts leftover in the documentation files

I believe we're going to feature freeze `release-3-0-0` soon so we shouldn't have to do this anymore going forward, hopefully

**Comment by jack1902** [comment]
I don't know how to remove these merge conflicts hence the `git merge -X theirs` to set `release3-1-0` commit history to look like the branch I did the rebase on and resolved these conflicts

File: docs/source/dynamic-outputs.rst [hunk 1]
```diff
@@ -0,0 +1,129 @@
+Dynamic Outputs
```
Comment (ryandeivert) [inline docs/source/dynamic-outputs.rst]: this format is now outdated and should be updated for consistency (either here or in another PR)

Comment (jack1902) [reply docs/source/dynamic-outputs.rst]: Will update and push this (I'll squash the commit into the one that introduced this)

**Comment by Ryxias** [comment]
This PR has been superseded by https://github.com/airbnb/streamalert/pull/1136

**Pull request closed without merging**